src/Cooked/MockChain/GenerateTx.hs

```diff
@@ -226,15 +217,48 @@ txOutRefsToTxInsReference =
     )
     . mapM Ledger.toCardanoTxIn
 
--- Convert a list of 'Api.TxOutRef' into a 'Cardano.TxInsCollateral'
-txOutRefsToTxSkelInsCollateral :: [Api.TxOutRef] -> TxGen (Cardano.TxInsCollateral Cardano.ConwayEra)
-txOutRefsToTxSkelInsCollateral =
-  throwOnToCardanoError "txOutRefsToTxInCollateral"
-    . fmap toTxInsCollateral
-    . mapM Ledger.toCardanoTxIn
-  where
-    toTxInsCollateral [] = Cardano.TxInsCollateralNone
-    toTxInsCollateral ins = Cardano.TxInsCollateral Cardano.AlonzoEraOnwardsConway ins
+-- Computes the collateral triplet from the fee and the collateral inputs in
+-- the context
+toCollateralTriplet ::
+  TxGen
+    ( Cardano.TxInsCollateral Cardano.ConwayEra,
+      Cardano.TxTotalCollateral Cardano.ConwayEra,
+      Cardano.TxReturnCollateral Cardano.CtxTx Cardano.ConwayEra
+    )
+toCollateralTriplet = do
```

Comment:
I think this implementation deserves comments to describe the main steps here.
There are non trivial things happening.

Reply:
Ok, I will comment it line by line.